client/state/billing-data/util.js

```diff
@@ -0,0 +1,18 @@
+/**
+ * External dependencies
+ */
+import { assign } from 'lodash';
+import { moment } from 'i18n-calypso';
+
+/**
+ * Parses the date within a transaction.
+ * Returns a copy of the updated transaction.
+ *
+ * @param  {Object}  transaction  Transaction object
+ * @return {Object}               Updated transaction with parsed date
+ */
+export const parseDate = ( transaction ) => {
+	return assign( {}, transaction, {
+		date: moment( transaction.date ).toDate()
```

Comment:
Good catch, I didn't know about this incompatibility. I've moved them to the selectors, it makes most sense to me.